### DISCUSSION TITLE
withAllTags() and withAnyTags() support slugs with hyphens?

### DISCUSSION THREAD

**Discussion by bartclaeys** [spatie/laravel-tags#437] [Q&A]
Are `withAllTags()` and `withAnyTags()` supposed to work with slugs as well, eg. `foo-bar`?

In my experience they do not, so I have to convert hyphens into spaces before using these.

**Comment by freekmurze** [comment]
They should work. If not, please submit a failing test (and possibly a fix) to demonstrate the problem.

**Comment by falconsmilie** [comment]
I have a similar issue.

This call here; 

https://github.com/spatie/laravel-tags/blob/b21031052aa12b6b53130b0b05d5fe54489167fa/src/HasTags.php#L76 

is returning a `Collection` where `items` contains `[0 => null]` when querying for tags by slug. With or without `type` yields the same result.

Usage:
```
MyModel::withAllTags('myTag', 'myType')->get()
```

When querying a regular string it returns with an array of `Tag`.

composer.json excerpt
```
        "php": "^8.2",
        ...
        "filament/filament": "^2.0",
        "filament/notifications": "^2.0",
        "filament/spatie-laravel-tags-plugin": "^2.0",
        ...
        "laravel/framework": "^9.19",
        ...
```

**Comment by falconsmilie** [comment]
@bartclaeys https://github.com/spatie/laravel-tags/pull/443
